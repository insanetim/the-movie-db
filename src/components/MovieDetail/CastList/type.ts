import { ICast } from 'src/interfaces/movie.interface'

export interface CastListProps {
  cast: ICast[]
}
