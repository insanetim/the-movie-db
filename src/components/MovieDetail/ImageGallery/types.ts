import { IMovieImage } from 'src/interfaces/movie.interface'

export type ImageGalleryProps = {
  images: IMovieImage[]
  title: string
}
