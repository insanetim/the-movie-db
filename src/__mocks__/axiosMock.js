function mockAxios() {
  const request = jest.fn()
  const mockedInterceptorsManager = {
    use: jest.fn(() => ({ type: 'interceptor-usage-token' })),
    eject: jest.fn()
  }
  const mockedProxyClient = Object.assign(request, {
    request,
    interceptors: {
      request: mockedInterceptorsManager,
      response: mockedInterceptorsManager
    },
    get: jest.fn((url, config) => request({ ...config, url, method: 'get' })),
    delete: jest.fn((url, config) => request({ ...config, url, method: 'delete' })),
    head: jest.fn((url, config) => request({ ...config, url, method: 'head' })),
    options: jest.fn((url, config) => request({ ...config, url, method: 'options' })),
    post: jest.fn((url, config) => request({ ...config, url, method: 'post' })),
    put: jest.fn((url, data, config) => request({ ...config, url, method: 'put', data })),
    patch: jest.fn((url, config) => request({ ...config, url, method: 'patch' }))
  })

  return mockedProxyClient
}

export default mockAxios
