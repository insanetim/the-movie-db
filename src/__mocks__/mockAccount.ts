import { IAccount } from 'src/interfaces/account.interface'

const mockAccount: IAccount = {
  avatar: {
    gravatar: {
      hash: '/image'
    },
    tmdb: {
      avatar_path: null
    }
  },
  id: 1234,
  include_adult: true,
  iso_639_1: 'en',
  iso_3166_1: 'en',
  name: '',
  username: 'test/username'
}

export default mockAccount
