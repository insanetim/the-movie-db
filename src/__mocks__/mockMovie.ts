import type { IBackdrop, ICast, ICrew, IMovie, IMovieDetailExtended } from 'src/interfaces/movie.interface'

export const mockMovie: IMovie = {
  adult: false,
  backdrop_path: '/image',
  genre_ids: [1, 2, 3],
  id: 123,
  original_language: 'en',
  original_title: 'test/title',
  overview: 'test/overview',
  popularity: 123,
  poster_path: '/image',
  release_date: '01-01-1999',
  title: 'test/title',
  video: false,
  vote_average: 123,
  vote_count: 123
}

export const mockMovieDetail: IMovieDetailExtended = {
  accountStates: { favorite: false, id: 123, rated: false, watchlist: false },
  adult: false,
  backdrop_path: '/image',
  belongs_to_collection: null,
  budget: 1000000,
  credits: {
    cast: [
      {
        adult: true,
        cast_id: 123,
        character: 'test/character',
        credit_id: '123',
        gender: null,
        id: 123,
        known_for_department: '',
        name: 'test/name',
        order: 123,
        original_name: 'test/name',
        popularity: 123,
        profile_path: '/image'
      }
    ],
    crew: [
      {
        adult: true,
        credit_id: '123',
        department: 'test/department',
        gender: null,
        id: 123,
        job: 'test/job',
        known_for_department: '',
        name: 'test/name',
        original_name: 'test/name',
        popularity: 123,
        profile_path: '/image'
      }
    ],
    id: 123
  },
  genre_ids: [1, 2, 3],
  genres: [{ id: 1, name: 'test/genre' }],
  homepage: null,
  id: 123,
  images: [
    {
      aspect_ratio: 0.5625,
      file_path: '/image',
      height: 900,
      iso_639_1: null,
      vote_average: 123,
      vote_count: 123,
      width: 1600
    }
  ],
  imdb_id: null,
  original_language: 'en',
  original_title: 'test/title',
  overview: 'test/overview',
  popularity: 123,
  poster_path: '/image',
  production_companies: [{ id: 123, logo_path: null, name: 'test/company', origin_country: 'USA' }],
  production_countries: [{ iso_3166_1: 'en', name: 'USA' }],
  release_date: '01-01-1999',
  revenue: 3000000,
  runtime: 150,
  spoken_languages: [{ iso_639_1: 'en', name: 'EN' }],
  status: 'test/status',
  tagline: null,
  title: 'test/title',
  video: false,
  vote_average: 123,
  vote_count: 123
}

export const mockCast: ICast = {
  adult: true,
  cast_id: 123,
  character: 'test/character',
  credit_id: '123',
  gender: null,
  id: 123,
  known_for_department: '',
  name: 'test/name',
  order: 123,
  original_name: 'test/name',
  popularity: 123,
  profile_path: '/image'
}

export const mockCrew: ICrew = {
  adult: true,
  credit_id: '123',
  department: 'test/department',
  gender: null,
  id: 123,
  job: 'test/job',
  known_for_department: '',
  name: 'test/name',
  original_name: 'test/name',
  popularity: 123,
  profile_path: '/image'
}

export const mockImage: IBackdrop = {
  aspect_ratio: 0.5625,
  file_path: '/image',
  height: 900,
  iso_639_1: null,
  vote_average: 123,
  vote_count: 123,
  width: 1600
}
