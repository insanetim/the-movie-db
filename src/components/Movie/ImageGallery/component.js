import PropTypes from 'prop-types'
import { Row, Col, Carousel } from 'antd'

const ImageGallery = ({ images }) => (
  <Row>
    <Col span={24}>
      <Carousel autoplay>
        {images.map(image => (
          <div
            key={image.file_path}
            className='movie-image'
          >
            <img
              src={`https://image.tmdb.org/t/p/original${image.file_path}`}
              alt=''
            />
          </div>
        ))}
      </Carousel>
    </Col>
  </Row>
)

ImageGallery.propTypes = {
  images: PropTypes.arrayOf(
    PropTypes.shape({
      file_path: PropTypes.string
    })
  )
}

ImageGallery.defaultProps = {
  images: []
}

export default ImageGallery
