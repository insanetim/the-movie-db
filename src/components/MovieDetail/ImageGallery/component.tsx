import { Carousel, Col, Row } from 'antd'

import { ImageGalleryProps } from './types'

const ImageGallery: React.FC<ImageGalleryProps> = ({ images, title }) => (
  <Row>
    <Col span={24}>
      <Carousel autoplay>
        {images.map(image => (
          <div
            className='movie-image'
            key={image.file_path}
          >
            <img
              alt={title}
              src={`https://image.tmdb.org/t/p/original${image.file_path}`}
            />
          </div>
        ))}
      </Carousel>
    </Col>
  </Row>
)

export default ImageGallery
