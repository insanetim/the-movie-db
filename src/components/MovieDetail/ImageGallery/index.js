import ImageGallery from './component'

export default ImageGallery
