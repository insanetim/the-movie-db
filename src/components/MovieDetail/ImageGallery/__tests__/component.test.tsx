import { render } from '@testing-library/react'
import { mockImage } from 'src/__mocks__/mockMovie'

import type { ImageGalleryProps } from '../types'

import ImageGallery from '../component'

describe('ImageGallery component', () => {
  it('matches snapshot', () => {
    const props: ImageGalleryProps = {
      images: [mockImage],
      title: 'test/title'
    }
    const { asFragment } = render(<ImageGallery {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
