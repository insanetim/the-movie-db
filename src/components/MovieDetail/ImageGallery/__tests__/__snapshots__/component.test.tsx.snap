// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageGallery component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-row css-dev-only-do-not-override-6j9yrn"
  >
    <div
      class="ant-col ant-col-24 css-dev-only-do-not-override-6j9yrn"
    >
      <div
        class="ant-carousel css-dev-only-do-not-override-6j9yrn"
      >
        <div
          class="slick-slider slick-initialized"
        >
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="movie-image"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                  >
                    <img
                      alt="test/title"
                      src="https://image.tmdb.org/t/p/original/image"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
